P202.Training.Test/RoleServiceTest.cs

```diff
@@ -0,0 +1,128 @@
+﻿using AutoMapper;
+using Moq;
+using P202.Training.Data.Entities;
+using P202.Training.Data.Repositories;
+using P202.Training.Domain;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace P202.Training.Test
+{
+    public class RoleServiceTest
+    {
+        [Fact]
+        public void RoleService_CreateData()
+        {
+            // Arrange
+            Mock<IRoleRepository> roleRepository = new Mock<IRoleRepository>();
+            Mock<IMapper> mappingService = new Mock<IMapper>();
+
+            var newRole = new Role
+            {
+                Id = 1,
+                Name = "Admin"
+            };
+
+            roleRepository.Setup(r => r.CreateRole(newRole));           
+
+            mappingService.Setup(m => m.Map<Role, Role>(It.IsAny<Role>()))
+                            .Returns(newRole);          
+
+            // Act
+            roleRepository.Object.CreateRole(newRole);
```

Comment:
Really Minor: I'd assign `roleRepository.Object` to a variable and use it instead of referencing the internal property `Object` everytime.

Reply:
You are right! Done!